tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/IbmMq/IbmMqHeadersAdapter.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="IbmMqHeadersAdapter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Datadog.Trace.Headers;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.IbmMq;
+
+internal readonly struct IbmMqHeadersAdapter : IHeadersCollection
+{
+    private static readonly string[] EmptyValue = { };
+    private readonly IMqMessage _message;
+
+    public IbmMqHeadersAdapter(IMqMessage message)
+    {
+        _message = message;
+    }
+
+    /// <summary>
+    /// Used to normalize the property name, since IBM MQ only allows names which are valid Java identifiers
+    /// </summary>
+    /// <param name="name">Originally provided value</param>
+    /// <returns>Normalized name</returns>
+    private static string NormalizeName(string name)
+    {
+        var sb = new StringBuilder();
+        foreach (var c in name)
+        {
+            sb.Append(c is >= 'a' and <= 'z' or >= '0' and <= '9' ? c : '_');
+        }
+
+        var result = sb.ToString();
+        return result;
```

Comment:
You may want to use the `StringBuilderCache` here to avoid allocating a `StringBuilder` every time they consume a message. Also, you know the size of the string, so can be efficient about it!

```suggestion
        var sb = StringBuilderCache.Acquire(name.Length);
        foreach (var c in name)
        {
            sb.Append(c is >= 'a' and <= 'z' or >= '0' and <= '9' ? c : '_');
        }

        return StringBuilderCache.GetStringAndRelease(sb);
```